packages/service-worker/src/low_level.ts

```diff
@@ -37,6 +41,46 @@ export interface UpdateActivatedEvent {
   current: {hash: string, appData?: Object};
 }
 
+/**
+ * An event emitted when a new version of the app has been detected but downloaded yet.
```

Comment:
```suggestion
 * An event emitted when a new version of the app has been detected but not downloaded yet.
```

Or (to make it clear that they don't need to do anything to start the download) maybe even:

```suggestion
 * An event emitted when the service worker has detected a new version of the app on the server and is about to start downloading it.
```